### ISSUE TITLE
Request for a comment feature Like Adobe Reader

### ISSUE THREAD

**Issue by sornay** [xournalpp/xournalpp#5624]
### Is Your Feature Request Related to a Problem?

Currently, Xournal++ lacks a dedicated feature for adding comments. When providing detailed feedback or annotations, there may not be enough space within the PDF itself to accommodate large amounts of text. For students working with PDF documents containing questions and answers, there is no easy way to hide the answers while studying.

### Feature Description

I would like to request a comment feature similar to Adobe Reader's commenting system in Xournal++. This feature would allow users to add sticky notes, text boxes, or other types of annotations directly on PDF documents. The comments should be easily identifiable and distinguishable from the original document content.

Furthermore, it would be beneficial if the commenting system could handle large amounts of text when necessary. In cases where there is limited space within the PDF itself, the commenting feature should provide a way to expand the comment area or create a pop-up window to accommodate longer comments or notes without obstructing the document's content.

Additionally, it would be useful if the comments could be color-coded, highlighted, or have customizable styles to make them more visually appealing and organized. The ability to reply to existing comments or create threads could also be helpful for facilitating discussions or addressing specific points within the document.

**Comment by github-kp** [comment]
I am also missing at least a feature to create sticky notes / popup notes, that popup when clicking them. The popup window could be resizable and allows viewing the full text in a scrollable way. It should also be possible to edit the text in the popup window as well. When hovering a "minimized" popup note with the mouse, a tooltip of the text could be shown - at least for the first few lines of the full text.

**Comment by tobiasBora** [comment]
Agree… Would be great to support the native PDF annotation (while preserving them during import/export), including also line, highlights… (at least to preserve them from import to export) as they are widely supported by many pdf readers.

For now, you cannot even import + export a pdf with comments as it will basically rasterize the image (but the text is lost, and users won't be able to change them later). It would be great to:
- provide a way to preserve annotations existing in a PDF via import + export
- provide a way to create new native PDF annotations, notably comments that I'm really missing

This might also be related to https://github.com/xournalpp/xournalpp/issues/3127
